### PR TITLE
Testflight

### LINKED ISSUE #387: During CI, publish iOS app to TestFlight
This setup exists on `sentry-cocoa` and is useful to do some "manual testing" before a release without the need to build locally and run on a device.

A failed attempt here: https://github.com/getsentry/sentry-dart/pull/340

Potential to extra the work done on https://github.com/getsentry/sentry-cocoa into a git repo that can be added via gittree or submodule to be reused/maintained across repos. We'd like to see this in also with other repos such as Unity, Flutter, Cordova, React Native and Xamarin.

### PR DESCRIPTION
**Pull request by denrase**
#skip-changelog

## :scroll: Description

Setup an action to upload the iOS sample app to TestFlight.

## :bulb: Motivation and Context

Closes #387

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [ ] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

- [ ] Don't run action on PR push
- Since we bumped to iOS 12, we need to wait for v8 to merge this PR

### CODE REVIEW

**Comment by denrase** [comment]
@kahest @buenaflor We need to create the app in AppStore Connect, add a provisioning profile to the match repo, and add the secrets to the repository that we also have in the sentry-cocoa repo so we can upload the build. I don't have permission to do any of these.

**Comment by denrase** [comment]
- [x] Provisioning profile needs to be created in the match repo.
- [x] Someone with permissions needs to check if all secrets are set in the repository.

**Comment by kahest** [comment]
@denrase thanks! secrets are set

**Comment by denrase** [comment]
@kahest @buenaflor We need to add the push notification capability to the provisioning profile and update it in the match repo.

**Comment by denrase** [comment]
@buenaflor Any update here?

**Comment by denrase** [comment]
@buenaflor The PP works now, thank you! I think we have one more issue. I guess the token we use here is missing a permission?

```
☑ uploading source maps                                                             

INFO    2024-02-20 14:54:09.615728 +00:00 sentry-cli was invoked with the following command line: "/Users/runner/work/sentry-dart/sentry-dart/flutter/example/.dart_tool/pub/bin/sentry_dart_plugin/sentry-cli" "--log-level" "info" "releases" "--org" "sentry-sdks" "--project" "sentry-flutter" "set-commits" "sentry_flutter_example@7.16.1" "--auto" "--ignore-missing"

error: API request failed
  caused by: sentry reported an error: You do not have permission to perform this action. (http status: 403)

exitCode: 1
```

**Comment by denrase** [comment]
- [x] Disable source mode uploads to check if this a sentry issue

File: flutter/example/ios/Flutter/AppFrameworkInfo.plist [hunk 1]
```diff
@@ -21,6 +21,6 @@
 	<key>CFBundleVersion</key>
 	<string>1.0</string>
 	<key>MinimumOSVersion</key>
-	<string>11.0</string>
+	<string>12.0</string>
```
Comment (buenaflor) [inline flutter/example/ios/Flutter/AppFrameworkInfo.plist:24]: Is this intentional? since we updated the min version to 12 in v8

Comment (denrase) [reply flutter/example/ios/Flutter/AppFrameworkInfo.plist:24]: I will try again if building/ci works with 11

Comment (denrase) [reply flutter/example/ios/Flutter/AppFrameworkInfo.plist:24]: Only wirks with v12. I created a new PR based on the V8 branch.

**Comment by denrase** [comment]
Closed in favour of #1938

**Pull request closed without merging**